src/Metadata/ConstraintsTrait.php

```diff
@@ -127,13 +129,8 @@ protected static function getKeyConstraints(): Collection
     {
         return new Collection([
             'keyid_hash_algorithms' => [
-                new Count(['min' => 1]),
-                new Type(['type' => 'array']),
-              // The keys for 'hashes is not know but they all must be strings.
-                new All([
-                    new Type(['type' => 'string']),
-                    new NotBlank(),
-                ]),
+                new Optional(),
```

Comment:
Why are we adding the `Optional` constraint here? If I understand this syntax correctly, previously we expected it to be a non-empty array. Wouldn't making it optional break that?

Reply:
per this issue https://github.com/theupdateframework/tuf/issues/848
`keyid_hash_algorithms` is not going in the JSON metadata files in the future so it should be optional. But currently this data reflects the setting in the `https://github.com/secure-systems-lab/securesystemslib` library that the Python TUF reference implementation uses. Our library(and probably the python library) is built to expect this setup. So if for some reason, which is very unlikely, `securesystemslib` was set up with different hash algorthims we should fail.

Reply:
Okay, that makes sense. Can you add a comment explaining why it's optional?

Reply:
Yep just pushed up a comment